### DISCUSSION TITLE
New NVDA+SHIFT+B message feels too verbose

### DISCUSSION THREAD

**Discussion by Mohamed00** [nvaccess/nvda#14213] [General]
Hi, I don't think I quite like the new message in 2022.4 when pressing NVDA+Shift+b. While it does report the same message consistently, I feel like the old message was better. Part of my issue with this change is that when plugging in or unplugging a charger, and when checking power status, I'm looking for different things. When a charger is plugged in, I'd like to know that the battery is being charged and its percentage. However, when pressing NVDA+shift+b, I'd like to hear the battery percentage first, then how much time is left or a charging message if the battery is being recharged, because when pressing NVDA+Shift+b, I'm looking for the battery percentage. Having AC disconnected announced before the battery message feels a bit too verbose for me. Thoughts?

**Comment by Brian1Gaff** [comment]
Where is this message being generated, in Windows or by NVDA from the status 
it sees?
 Brian

-- 
***@***.***
Sent via blueyonder.(Virgin media)
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
----- Original Message ----- 
From: "Mohamed Al-Hajamy" ***@***.***>
To: "nvaccess/nvda" ***@***.***>
Cc: "Subscribed" ***@***.***>
Sent: Tuesday, October 04, 2022 3:12 PM
Subject: [nvaccess/nvda] New NVDA+SHIFT+B message feels too verbose 
(Discussion #14213)

**Reply by seanbudd** [reply]
It's generated by NVDA. We recently changed how the message was presented in https://github.com/nvaccess/nvda/pull/14035, which is staged for 2022.4

**Comment by bdorer** [comment]
I second this. Normally a user knows wheather AC is disconnected or not.

**Comment by seanbudd** [comment]
Please create an issue and we will address this for 2022.4.

Use the feature request template or just a summary of the issue for the issue description.

When battery status changes, do you think the battery percentage should be said first too, or should the current behaviour of announcing the AC state first remain?

The original intention was to make the messages consistent, but if that's not a good user experience we can change the order based on use case.

**Comment by Mohamed00** [comment]
I created #14214. When plugging in a charger? Yes, I think this can remain the same, because in that case, I feel like it makes sense in that case to say that the battery is charging first and then the percentage.

**Reply by seanbudd** [reply]
Can you add that note to the issue?

Thanks.

**Comment by XLTechie** [comment]
While not a big issue for me, I agree with all of Mohamed's points about the 
messages in question. The priorities of information needed are different in the 
situations of plug/unplug, and pressing the battery status key.

**Comment by Brian1Gaff** [comment]
Yes having just listened on a laptop, it just feels wrong, but cannot 
actually say quite why!
 Brian

-- 
***@***.***
Sent via blueyonder.(Virgin media)
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
----- Original Message ----- 
From: "Luke Davis" ***@***.***>
To: "nvaccess/nvda" ***@***.***>
Cc: "Brian Gaff" ***@***.***>; "Comment" 
***@***.***>
Sent: Wednesday, October 05, 2022 1:50 AM
Subject: Re: [nvaccess/nvda] New NVDA+SHIFT+B message feels too verbose 
(Discussion #14213)
